java/src/test/java/io/grpc/examples/wallet/ObservabilityTest.java

```diff
@@ -0,0 +1,161 @@
+package io.grpc.examples.wallet;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import io.grpc.ManagedChannel;
+import io.grpc.Server;
+import io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder;
+import io.grpc.netty.shaded.io.grpc.netty.NettyServerBuilder;
+import io.grpc.stub.StreamObserver;
+import io.opencensus.common.Duration;
+import io.opencensus.contrib.grpc.metrics.RpcViews;
+import io.opencensus.exporter.metrics.util.IntervalMetricReader;
+import io.opencensus.exporter.metrics.util.MetricExporter;
+import io.opencensus.exporter.metrics.util.MetricReader;
+import io.opencensus.metrics.Metrics;
+import io.opencensus.metrics.export.Metric;
+import io.opencensus.metrics.export.MetricDescriptor;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.export.SpanData;
+import io.opencensus.trace.export.SpanExporter;
+import io.opencensus.trace.samplers.Samplers;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ObservabilityTest {
+
+  private static final int EXPORT_WAIT_SEC = 15;
+
+  private static final Set<String> METRICS =
+      new HashSet<>(
+          Arrays.asList(
+              "grpc.io/client/roundtrip_latency",
+              "grpc.io/client/completed_rpcs",
+              "grpc.io/client/sent_bytes_per_rpc",
+              "grpc.io/client/received_bytes_per_rpc",
+              "grpc.io/client/sent_messages_per_rpc",
+              "grpc.io/client/received_messages_per_rpc",
+              "grpc.io/server/completed_rpcs",
+              "grpc.io/server/sent_bytes_per_rpc",
+              "grpc.io/server/received_bytes_per_rpc",
+              "grpc.io/server/sent_messages_per_rpc",
+              "grpc.io/server/received_messages_per_rpc",
+              "grpc.io/server/server_latency"));
+
+  private Server server;
+  private ManagedChannel channel;
+  private RecordingMetricsExporter recordingMetricsExporter;
+  private RecordingTraceExporter recordingTraceExporter;
+
+  private static class WalletImpl extends WalletGrpc.WalletImplBase {
```

Comment:
nit: put inner classes in the bottom (although not explicitly specified in Java style guide, most people follows the convention with the _principle of least surprise_ and consider inner classes are less important ([SO link](https://stackoverflow.com/questions/20180533/what-is-the-convention-for-the-placement-of-inner-classes-in-java))).